gcloud/dns/client.py

```diff
@@ -108,3 +108,29 @@ def zone(self, name, dns_name):
         :returns: a new ``ManagedZone`` instance
         """
         return ManagedZone(name, dns_name, client=self)
+
+    def get_zone(self, zone_name):
+        """Get a zone by name.
+
+        If the zone isn't found, this will raise a
+        :class:`gcloud.exceptions.NotFound`.
+
+        For example::
+
+          >>> try:
+          >>>   zone = client.get_zone('my-zone')
```

Comment:
I'd generally lean towards raising `NotFound`, because situations where the user will not be expecting the failure and will not check for it are common. On the other hand, a user who does feel the need to handle error conditions will need to handle several others besides `NotFound`.

Providing an alternative `lookup_thing` returning `None` can be useful, but whether it is really worthwhile will depend on the API. In my opinion, the main advantage is in keeping the client code independent of the exception class when the code needs to check for `NotFound`.